exp/controllers/azuremachinepool_controller.go

```diff
@@ -297,12 +298,15 @@ func (r *AzureMachinePoolReconciler) reconcileNormal(ctx context.Context, machin
 	switch machinePoolScope.ProvisioningState() {
```

Comment:
not part of this pr, but we should probably move this to machinepoolscope.

Reply:
#819 does a lot of rework of this area so didn't want to conflict too much, but yes agreed